plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,415 @@
+  type: str
+  sample: "Success"
+
+mr:
+  description: API object.
+  returned: success
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.common.text.converters import to_native, to_text
+
+from ansible_collections.community.general.plugins.module_utils.version import LooseVersion
+from ansible_collections.community.general.plugins.module_utils.gitlab import (
+    auth_argument_spec, gitlab_authentication, gitlab, ensure_gitlab_package
+)
+
+
+class GitlabMergeRequest(object):
+
+    def __init__(self, module, project, gitlab_instance):
+        self.repo = gitlab_instance
+        self._module = module
+        self.project = self.get_project(project)
+
+    '''
+    @param project Name of the project
+    '''
+    def get_project(self, project):
+        try:
+            return self.repo.projects.get(project)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to get the project: %s" % to_native(e))
+
+    '''
+    @param branch Name of the branch
+    '''
+    def get_branch(self, branch):
+        try:
+            return self.project.branches.get(branch)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to get the branch: %s" % to_native(e))
+
+    '''
+    @param title Title of the Merge Request
+    @param source_branch Merge Request's source branch
+    @param target_branch Merge Request's target branch
+    @param state_filter Merge Request's state to filter on
+    '''
+    def get_mr(self, title, source_branch, target_branch, state_filter):
+        mrs = []
+        try:
+            mrs = self.project.mergerequests.list(search=title, source_branch=source_branch, target_branch=target_branch, state=state_filter)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to list the Merge Request: %s" % to_native(e))
+
+        if len(mrs) > 1:
+            self._module.fail_json(msg="Multiple Merge Requests matched search criteria.")
+        if len(mrs) == 1:
+            try:
+                return self.project.mergerequests.get(id=mrs[0].iid)
+            except gitlab.exceptions.GitlabGetError as e:
+                self._module.fail_json(msg="Failed to get the Merge Request: %s" % to_native(e))
+
+    '''
+    @param username Name of the user
+    '''
+    def get_user(self, username):
+        try:
+            users = self.project.users.list(search=username, all=True)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to list the users: %s" % to_native(e))
+
+        for user in users:
+            if (user.username == username):
+                return user
+
+    '''
+    @param users List of usernames
+    '''
+    def get_user_ids(self, users):
+        return [self.get_user(user).id for user in users]
```

Comment:
`self.get_user(user).id` may raise an unexpected Exception if the user you're looking for does not exist.
For example :
- `assignee_ids` contains `john`
- The GitLab API would answer with a `johnny` user even if `john` does not exist (since `/users` API makes a fuzzy search, see [docs](https://docs.gitlab.com/ee/api/users.html#for-non-administrator-users))
- Your `get_user` function will then return `None`
- `self.get_user(user).id` will fail

I think it would be preferable to fail properly with `self._module.fail_json` in that case.